exporters/otlp/src/otlp_http_client.cc

```diff
@@ -896,7 +896,7 @@ OtlpHttpClient::createSession(
   // Parse uri and store it to cache
   if (http_uri_.empty())
   {
-    auto parse_url = opentelemetry::ext::http::common::UrlParser(std::string(options_.url));
+    const auto &parse_url = opentelemetry::ext::http::common::UrlParser(options_.url);
```

Comment:
Why change it to `const auto &`? In my understanding, `const auto &` can not refer to a temporary object here, because it's may be de destroyed after this sentense.
`T f = T(..);` will construct T only once and will not call copy/move constructor. So I think it's useless here.

Reply:
I made it _const_ because it is never modified later in function, and _reference_ as a future assurance that it doesn't create a copy, since copy elision is not guaranteed prior to C++17.

I believe it is not a problem with temporary in this case because the [lifetime extension](https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary) would kick in to bind the prvalue to the scope of the function it is being used in. 

Although, I agree, it would've been a problem if we were to pass around the pointer to its `.data()` once the function scope ends. 

Seeing this could cause some confusion, I removed the reference to make the intent clearer.

Reply:
Thanks, my mistake